src/components/Auth/SignUp/SignUp.react.js

```diff
@@ -75,86 +84,113 @@ export default class SignUp extends Component {
+      isEmail = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
+      emailError = !(email && isEmail);
     } else if (event.target.name === 'password') {
-      password = event.target.value;
-      let validPassword = password.length >= 6;
-      state.passwordValue = password;
-      state.passwordError = !(password && validPassword);
-      if (validPassword) {
-        let result = zxcvbn(password);
-        state.passwordScore = result.score;
-      } else {
-        state.passwordScore = -1;
-      }
+      passwordValue = event.target.value;
+      validPassword = passwordValue.length >= 6;
+      passwordError = !(passwordValue && validPassword);
+      passwordConfirmError = !(
+        passwordValue === this.state.confirmPasswordValue
+      );
     } else if (event.target.name === 'confirmPassword') {
-      password = this.state.passwordValue;
-      confirmPassword = event.target.value;
-      let validPassword = confirmPassword === password;
-      state.confirmPasswordValue = confirmPassword;
-      state.passwordConfirmError = !(validPassword && confirmPassword);
+      confirmPasswordValue = event.target.value;
+      validPassword = confirmPasswordValue === passwordValue;
+      passwordConfirmError = !(validPassword && confirmPasswordValue);
     }
 
-    if (
-      !this.state.emailError &&
-      !this.state.passwordError &&
-      !this.state.passwordConfirmError
-    ) {
-      state.validForm = true;
+    if (emailError) {
+      emailErrorMessage = 'Enter a valid Email Address';
+    } else if (passwordError) {
+      emailErrorMessage = '';
+      passwordErrorMessage = 'Minimum 6 characters required';
+      passwordConfirmErrorMessage = '';
+      captchaVerifyErrorMessage = '';
+    } else if (passwordConfirmError) {
+      emailErrorMessage = '';
+      passwordErrorMessage = '';
+      passwordConfirmErrorMessage = 'Check your password again';
+      captchaVerifyErrorMessage = '';
+    } else if (!isCaptchaVerified) {
+      emailErrorMessage = '';
+      passwordErrorMessage = '';
+      passwordConfirmErrorMessage = '';
+      captchaVerifyErrorMessage = 'Please confirm you are a human';
     } else {
-      state.validForm = false;
+      emailErrorMessage = '';
+      passwordErrorMessage = '';
+      passwordConfirmErrorMessage = '';
+      captchaVerifyErrorMessage = '';
     }
 
-    this.setState(state);
-
-    if (this.state.emailError) {
-      this.emailErrorMessage = 'Enter a valid Email Address';
-    } else if (this.state.passwordError) {
-      this.emailErrorMessage = '';
-      this.passwordErrorMessage = 'Minimum 6 characters required';
-      this.passwordConfirmErrorMessage = '';
-    } else if (this.state.passwordConfirmError) {
-      this.emailErrorMessage = '';
-      this.passwordErrorMessage = '';
-      this.passwordConfirmErrorMessage = 'Check your password again';
+    if (!emailError && !passwordError && !passwordConfirmError) {
```

Comment:
Check if captcha is also verified or not. Only in that case the user should be able to signup/login.